compiler/src/iree/compiler/Codegen/Utils/Utils.cpp

```diff
@@ -143,8 +143,16 @@ bool hasUkernel(IREE::HAL::ExecutableTargetAttr targetAttr,
   StringRef enabledUkernelsStr = enabledUkernels->getValue();
   // Resolve `default`.
   if (enabledUkernelsStr == "default") {
-    // Current defaults implemented here. Could depend on targetAttr.
-    enabledUkernelsStr = "none";
+    // Current defaults implemented here.
+    // Enable the mmt4d ukernel on x86_64 and on arm_64, but not for now when
+    // SVE is present.
+    if (isX86_64(targetAttr) ||
+        (isAArch64(targetAttr) && !hasFeature(targetAttr, "+sve") &&
+         !hasFeature(targetAttr, "+sve2"))) {
```

Comment:
We should also add `+sme` to this list.

Btw, `+sve2` and `+sme` imply `+sve` at the hardware level. Not sure how IREE models that, but it's probably safer to list all 3 options.

Reply:
@banach-space , PR updated to also exclude SME.

Feature implications at hardware level are not directly relevant here since the individual feature flags are user-controlled.